official/utils/logs/logger.py

```diff
@@ -28,38 +28,57 @@
 import numbers
 import os
 import threading
+import uuid
 
+from absl import flags
 import tensorflow as tf
 from tensorflow.python.client import device_lib
 
 METRIC_LOG_FILE_NAME = "metric.log"
 BENCHMARK_RUN_LOG_FILE_NAME = "benchmark_run.log"
 _DATE_TIME_FORMAT_PATTERN = "%Y-%m-%dT%H:%M:%S.%fZ"
 
+FLAGS = flags.FLAGS
 
 # Don't use it directly. Use get_benchmark_logger to access a logger.
 _benchmark_logger = None
 _logger_lock = threading.Lock()
 
 
-def config_benchmark_logger(logging_dir):
+def config_benchmark_logger(flag_obj=None):
   """Config the global benchmark logger"""
   _logger_lock.acquire()
   try:
     global _benchmark_logger
-    if logging_dir:
-      _benchmark_logger = BenchmarkFileLogger(logging_dir)
-    else:
+    if not flag_obj:
+      flag_obj = FLAGS
+
+    if (not hasattr(flag_obj, 'benchmark_logger_type') or
+        flag_obj.benchmark_logger_type == 'BaseBenchmarkLogger'):
       _benchmark_logger = BaseBenchmarkLogger()
+    elif flag_obj.benchmark_logger_type == 'BenchmarkFileLogger':
```

Comment:
Do we still need the filelogger as an option? If we are not going to use it, why complicate this with a third option that we have to maintain? That seems like it would allow us to strip half the code here and above.

Reply:
yes. The file logger is still useful for small model so that we can save the bigquery upload quota. The streaming one will be useful for those long running benchmark.

Reply:
Why not always stream?

Reply:
Bigquery API support to have a usage quota, even we currently have small load, I think its still worthwhile to not spam the service. For small dataset like cifar10 which take 40 mins to finish, I would still prefer to upload the data after the run.